contributingGuides/STYLE.md

```diff
@@ -684,125 +574,9 @@ When writing a stateless component you must ALWAYS add a `displayName` property
 
 ## Stateless components vs Pure Components vs Class based components vs Render Props - When to use what?
 
-*_1. Stateless components: Used when you don't need to maintain state or use lifecycle methods._*
-
-In many cases, we create components that do not need to have a state, lifecycle hooks or any internal variables. In other words just a simple component that takes props and renders something presentational. But often times we write them as class based components which come with a lot of cruft (for e.g., it has a state, lifecycle hooks and it is a javascript class which means that React creates instances of it) that is unnecessary in many cases.
-
-A quote from the React documentation:
+Class components are DEPRECATED. Use function components and React hooks.
 
-> These components must not retain internal state, do not have backing instances, and do not have the component lifecycle methods. They are pure functional transforms of their input, with zero boilerplate. However, you may still specify .propTypes and .defaultProps by setting them as properties on the function, just as you would set them on an ES6 class.
-
-
- - Here is an [example](https://github.com/Expensify/JS-Libs/blob/e3b7ee4b111a3a370a1427ad904485df4e65a472/lib/components/StepProgressBar.js#L20) from our codebase of a stateless component.
-```js
-const StepProgressBar = ({steps, currentStep}) => {
-    const currentStepIndex = Math.max(0, _.findIndex(steps, step => step.id === currentStep));
-    return (
-        <div id="js_steps_progress" className="progress-wrapper">
-...
-
-```
-*_2. Pure components: Use to improve performance where a component does not need to be rendered too often._*
```

Comment:
This whole section is a bit out of date though I suppose we can mention something about performance. `React.memo()` is basically the replacement for any `PureComponent` or `shouldComponentUpdate()`.